draft-ietf-dnsop-svcb-https.md

```diff
@@ -1082,18 +1082,47 @@ single-client granularity.
 
 ## Interaction with Alt-Svc
 
-Clients that do not implement support for Encrypted ClientHello MAY
-skip the HTTPS RR query
-if a usable Alt-Svc value is available in the local cache.
-If Alt-Svc connection fails, these clients SHOULD fall back to the HTTPS RR
-client connection procedure ({{client-behavior}}).
-
-Clients that implement support for ECH MUST perform the HTTPS RR query first,
-and MUST only make use of Alt-Svc when operating in SVCB-optional mode (see
-{{ech-client-behavior}}).
-
-This specification does not alter the DNS records used when connecting
-to an Alt-Svc hostname (typically A and/or AAAA only).
+Clients that implement support for both Alt-Svc and HTTPS records SHOULD
+retrieve any HTTPS records for the Alt-Svc alt-authority, and ensure that
+their connection attempts are consistent with both the Alt-Svc parameters
+and any received HTTPS SvcParams.  If present, the HTTPS record's target
+and port override the alt-authority.  For example, suppose that
+"https://example.com" sends an Alt-Svc field value of:
+
+~~~ HTTP
+Alt-Svc: h2="alt.example:443", h2="alt2.example:443", h3=":8443"
+~~~
+
+The client would retrieve the following HTTPS records:
+
+    alt.example.              IN HTTPS 1 . alpn=h2,h3 ech=...
+    alt2.example.             IN HTTPS 1 . alpn=h3 ech=...
+    _8443._https.example.com. IN HTTPS 1 alt3.example. (
+        port=9443 alpn=h2,h3 ech=... )
+
+The client could then attempt an HTTP/3 connection to `alt3.example:9443`
+with ECH, or an HTTP over TLS connection to `alt.example:443` with ECH,
+as these options are consistent with both an Alt-Svc field value and its
+HTTPS record.  An ECH-capable client would not connect to `alt2.example`,
+as there is no possible connection that is consistent with both ALPN values
+("h2" from Alt-Svc, "h3" from SvcParams), and the "ech" SvcParam triggers
+SVCB-reliant client behavior ({{ech-client-behavior}}).  Operators SHOULD
```

Comment:
This sentence ("An ECH-capable client [...]") seems to contain two steps at once. It's also misleading because non-ECH clients can also have see Alt-Svc vs ServiceForm incompatibilities. There's a difference between using ServiceMode records vs. A/AAAA if the alt2.example fetch gives back another TargetName. How about replacing:
```
    alt2.example.             IN HTTPS 1 . alpn=h3 ech=...
```
with:
```
    alt2.example.             IN HTTPS 1 alt2b.example. alpn=h3 ech=...
```

And then replace this sentence with:

> The client could not connect to `alt2b.example` because there is no possible connection that is consistent with both ALPN values ("h2" from Alt-Svc, "h3" from SvcParams). A SVCB-reliant client could not use the `alt2.example` Alt-Svc entry, but an SVCB-optional client could instead connect to `alt2.example` without the use of ServiceMode records. In an ECH-capable client, the "ech" SvcParam triggers SVCB-reliant client behavior ({{ech-client-behavior}}).